CRM/Core/BAO/CustomGroup.php

```diff
@@ -1250,6 +1250,25 @@ public static function deleteGroup($group, $force = FALSE) {
     return TRUE;
   }
 
+  /**
+   * Delete a record from supplied params.
+   * API3 calls deleteGroup() which removes the related civicrm_value_X table.
+   * This function does the same for API4.
+   *
+   * @param array $record
+   *   'id' is required.
+   * @return CRM_Core_DAO
+   * @throws CRM_Core_Exception
+   */
+  public static function deleteRecord(array $record) {
+    $customGroup = \Civi\Api4\CustomGroup::get(FALSE)
+      ->addWhere('id', '=', $record['id'])
+      ->execute()
+      ->first();
+    parent::deleteRecord($record);
+    CRM_Core_BAO_SchemaHandler::dropTable($customGroup['table_name']);
```

Comment:
The concept makes sense but if feels strangely circular to have a function called from APIv4 then call APIv4. I think this would be simpler:

```suggestion
    $table = CRM_Core_DAO::getFieldValue(__CLASS__, $record['id'], 'table_name');
    parent::deleteRecord($record);
    CRM_Core_BAO_SchemaHandler::dropTable($table);
```